theano/tensor/tests/test_basic.py

```diff
@@ -3441,6 +3441,23 @@ def test_stack_scalar_make_vector_constant(self):
         assert len([n for n in topo if isinstance(n, type(self.join_op))]) == 0
         assert f.maker.fgraph.outputs[0].dtype == 'int64'
 
+    def test_stack_new_interface(self):
+        """Test the new numpy-like interface: stack(tensors, axis=0)."""
+        a = tensor.imatrix('a')
+        b = tensor.imatrix('b')
+        s1 = stack(a, b)
+        s2 = stack([a, b])
+        f = function([a, b], [s1, s2], mode=self.mode)
+        v1, v2 = f([[1, 2]], [[3, 4]])
+        self.assertTrue(v1.shape == v2.shape)
+        self.assertTrue(numpy.all(v1 == v2))
+        s3 = join(1, *[shape_padaxis(t, 1) for t in [a, b]])
```

Comment:
I'd rather stay safe and compute s3 with numpy so that any bugs in shape_padaxis could be catched. Otherwise s3 and s4 have same implementation and are thus both correct or both wrong with no alarm signal.